hathor/transaction/util.py

```diff
@@ -48,13 +51,11 @@ def unpack_len(n: int, buf: bytes) -> tuple[bytes, bytes]:
     return buf[:n], buf[n:]
 
 
-def get_deposit_amount(mint_amount: int) -> int:
-    settings = get_global_settings()
+def get_deposit_amount(settings: HathorSettings, mint_amount: int) -> int:
```

Comment:
Should this method receive the percentage and let the caller pass it?

Reply:
What's your reasoning? In my opinion, we shouldn't. This would transfer the responsibility of passing the correct percentage to the caller, who would have to know to always pass `settings.TOKEN_DEPOSIT_PERCENTAGE` anyway, being prone to errors as users could call this with incorrect values. Receiving settings instead, leaves this logic internal to the function and impossible to mistake.

Reply:
Got it. Ok. We can do it later anyways.